src/components/shared/MarkerContextMenu.js

```diff
@@ -170,11 +170,17 @@ class MarkerContextMenu extends PureComponent<Props> {
     }
   };
 
+  _hidingTimeout: TimeoutID | null = null;
+
   _onHide = () => {
-    this.props.setContextMenuVisibility(false);
+    this._hidingTimeout = setTimeout(() => {
+      this._hidingTimeout = null;
+      this.props.setContextMenuVisibility(false);
+    }, 100);
```

Comment:
Same remark as for the call node context menu, `100` is pretty arbitrary but this seems to work fine for me.

Also I tried to extract a shareable implementation, but because we use actions we need at least the `dispatch` function and this is a bit complex, and I thought this wasn't worth the trouble.

Reply:
Ah I see it now. My same comment applies from the other PR. At the very least I think this should have some good code comments.